lib/msf/core/exploit.rb

```diff
@@ -752,6 +760,10 @@ def is_payload_compatible?(name)
     # what not?
     return false if !compatible?(pi)
 
+    if defined?(@needs_cleanup) && @needs_cleanup && !pi.can_cleanup
```

Comment:
This should just be `if self.needs_cleanup && !pi.can_cleanup` once using the `attr_accessor` as above.